### ISSUE TITLE
[BUG] matrix_pairwise_distance with cosine return doubled the values

### ISSUE THREAD

**Issue by vinson2233** [talboger/fastdist#6]
fastdist version : 1.1.1
numba version : 0.53.1

# Problems : 
All values calculated from `matrix_pairwise_distance` with cosine similarity return the double of the actual values(self similarity have cosine similarity = 2 instead of 1)

# Code to reproduce the BUG : 
```python
import numpy as np
import fastdist
u = np.random.normal(0,1,(4,4))
fastdist.matrix_pairwise_distance(u, fastdist.cosine, "cosine", return_matrix=True)
```
will return
```python
array([[ 2.        ,  1.25691085, -0.60143221, -1.52995311],
       [ 1.25691085,  2.        , -1.82710944, -1.74097934],
       [-0.60143221, -1.82710944,  2.        ,  1.25862114],
       [-1.52995311, -1.74097934,  1.25862114,  2.        ]])
```
See that the cosine similarity have value from [-2,2] instead of [-1,1]

# Quick temporary fix 
Divide the return matrix by 2

# Additional Info : 
`fastdist.cosine(u[0],u[1])` produce same value as `np.sum(u[0]*u[1])/(np.linalg.norm(u[0])*np.linalg.norm(u[1]))`  
Using `fastdist.cosine_pairwise_distance(u,return_matrix = False)` still return the correct cosine similarity.  
But using `fastdist.cosine_pairwise_distance(u,return_matrix = False)` make all the values got doubled.

**Comment by vinson2233** [comment]
Ups, I just realized there are commits regarding this issue. I have tried the latest `cosine_pairwise_distance` and the returned value quite weird for me.
cosine_pairwise_distance(u,True) #Different u from previously mentioned. I take the snippet directly from https://github.com/talboger/fastdist/blob/6bc4ccff0c13e30afb1a1ac189d2e9dbe38ea239/fastdist/fastdist.py#L288
```
array([[ 1.        , -0.18823476, -0.26611453, -0.40293875],
       [-0.37646952,  1.        ,  0.2488919 ,  0.78641663],
       [-0.53222907,  0.4977838 ,  1.        ,  0.78021149],
       [-0.80587751,  1.57283326,  1.56042298,  1.        ]])
```

The lower triangle part still got the values doubled. But if I change the code from https://github.com/talboger/fastdist/blob/6bc4ccff0c13e30afb1a1ac189d2e9dbe38ea239/fastdist/fastdist.py#L333 
to 
```
out_mat = out_mat + out_mat.T
```
The returned value becomes correct.

**Comment by talboger** [comment]
Ah, good catch! Yes sorry about the initial bug; I was waiting to bump the package to a new version with some other changes as well, but have yet to complete them so might just put out a new one with the fixed cosine similarity. I'll fix this and hopefully push a new version soon!

**Comment by vinson2233** [comment]
Alright, thanks for the awesome package anyway.

**Comment by talboger** [comment]
Just published v1.1.2 that fixes this issue!
